### PR TITLE
Return Promises from generateRequest

### PR DESCRIPTION
**Pull request by ryantmer**
iron-ajax supports a Promise-based API (although it's not really documented?), which makes doing multiple sequential requests much more fluent.

One slight weirdness is that for the framed request, we are forced to return the .completes, whereas in the non-framed case it works fine with just returning the result of generateRequest()... not sure why that is, but I've made both the `d2l-ajax` and `d2l-ajax-framed` use the same interface. It's slightly inconsistent with how `iron-ajax` itself is doing things, but it's also slightly nicer than having to remember to `request.completes.then` when using it.

Tested this with My Courses as well as Parent Portal UI and it works fine, so seems good for just a minor release.

### CODE REVIEW

**Comment by omsmith** [comment]
Given this wasn't exactly correct in terms of being a drop-in shim for iron-ajax before, and this isn't any less wrong, but more useful than before, LGTM

File: test/d2l-ajax/d2l-ajax.js [hunk 1]
```diff
@@ -200,6 +200,20 @@ describe('d2l-ajax', function() {
 			component._resetAuthTokenCaches();
 		});
 
+		it('should return a Promise', function(done) {
```
Comment (CodeBaboon) [inline test/d2l-ajax/d2l-ajax.js]: Is it worth testing to make sure that promise resolves properly when the request completes?

Comment (CodeBaboon) [inline test/d2l-ajax/d2l-ajax.js]: Oh wait I guess it does. Nvm.

Comment (ryantmer) [inline test/d2l-ajax/d2l-ajax.js]: Whoops, jumped the gun on the merge there, sorry! But yeah, the `done` takes care of that :)

**Pull request merged**